likecoin/admin/matters.php

```diff
@@ -352,6 +352,64 @@ function likecoin_post_attachment_to_matters( $attachment_id ) {
 	return $matters_attachment_id;
 }
 
+/**
+ * Upload a file as draft image to matters
+ *
+ * @param int|     $image_url image url to be uploaded to matters.
+ * @param object | $image_infos all images' information of the post.
+ */
+function likecoin_post_url_image_to_matters( $image_url, $image_infos ) {
+	global $post;
+	$post_id              = $post->ID;
+	$file_path            = $image_url;
+	$file_path_components = preg_split( '#[.]#', $file_path );
+	$file_mime_type       = 'image/' . end( $file_path_components );
+	$filename             = basename( $file_path );
+	$parent_post          = $post_id;
+	$matters_info         = get_post_meta( $parent_post, LC_MATTERS_INFO, true );
+	if ( ! $matters_info ) {
+		$matters_info = array(
+			'type' => 'post',
+		);
+	}
+	if ( isset( $matters_info['published'] ) && $matters_info['published'] ) {
+		return;
+	}
+	$matters_draft_id = isset( $matters_info['draft_id'] ) ? $matters_info['draft_id'] : null;
+	if ( ! $matters_draft_id ) {
+		if ( ! likecoin_get_admin_errors() ) {
+			likecoin_handle_matters_api_error( 'Cannot save draft before publishing' );
+		}
+		return;
+	}
```

Comment:
I think since we run this function first before `likecoin_save_to_matters`, on the first save this would always return false since matters_draft_id does not exists before saving. This would also break if user does not enable save draft but have enabled publish to matters.
Maybe we should call `likecoin_upload_url_image_to_matters` inside `likecoin_save_to_matters` instead?